apps/web/pages/event-types/[type]/index.tsx

```diff
@@ -132,6 +134,37 @@ export type FormValues = {
   bookerLayouts: BookerLayoutSettings;
 };
 
+interface DiffObject {
+  [key: string]: any;
+}
+
+function getObjectDifference(obj1: DiffObject, obj2: DiffObject): DiffObject {
+  const diff: DiffObject = {};
+
+  for (const key in obj1) {
+    if (obj1?.hasOwnProperty(key)) {
+      if (!obj2?.hasOwnProperty(key)) {
+        diff[key] = obj1[key];
+      } else if (typeof obj1[key] === "object" && typeof obj2[key] === "object") {
+        const nestedDiff = getObjectDifference(obj1[key], obj2[key]);
+        if (Object.keys(nestedDiff).length > 0) {
+          diff[key] = nestedDiff;
+        }
+      } else if (obj1[key] !== obj2[key]) {
+        diff[key] = obj1[key];
+      }
+    }
+  }
+
+  for (const key in obj2) {
+    if (obj2?.hasOwnProperty(key) && !obj1?.hasOwnProperty(key)) {
+      diff[key] = obj2[key];
+    }
+  }
+
+  return diff;
+}
```

Comment:
I havent tested but can we not use something like

const dirtyValues = watch({ dirtyFields: true });
from RHF to test for this?

Reply:
I can't understand what you are saying :(

Reply:
@Krishna2323 check this https://react-hook-form.com/docs/useformstate. React hook form gives us info about what fields have been changed in the form.